### PR TITLE
[fix] [broker] topics failed to delete after remove cluster from replicated clusters set and caused OOM

### PR DESCRIPTION
**Pull request by poorbarcode**
#### Background
- Users may start 2 clusters with the Geo-Replication feature with Global ZK
- Users want to hire one cluster, so remove the cluster from `namespace-level replicatedClusters`
  - Broker will remove the topics of the cluster removed automatically.
- Issue occurs:
  - The system topic `__change_events` being deleted first.
  - User topics can not be deleted anymore, because the operation `delete topic-level policies` can not be executed successfully anymore.
  - The task `topic.checkReplication` and other operations will retry again and again, as a result, the broker will crash of OOM  

---

Screenshot 2024-09-27 at 17 17 42

---

Screenshot 2024-09-27 at 17 18 56

### Modifications
- Skip removing topic-level policies if the system topic `__change_event` has been deleted.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: x

### CODE REVIEW

File: bin/ledgers.md
Comment (lhotari) [inline bin/ledgers.md:1]: delete?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java [hunk 1]
```diff
@@ -133,7 +133,13 @@ public CompletableFuture<Void> deleteTopicPoliciesAsync(TopicName topicName) {
         if (NamespaceService.isHeartbeatNamespace(topicName.getNamespaceObject()) || isSelf(topicName)) {
             return CompletableFuture.completedFuture(null);
         }
-        return sendTopicPolicyEvent(topicName, ActionType.DELETE, null);
+        return pulsarService.getNamespaceService().checkTopicExists(NamespaceEventsSystemTopicFactory
+                .getEventsTopicName(topicName.getNamespaceObject())).thenCompose(topicExistsInfo -> {
+            if (topicExistsInfo.isExists()) {
+                return CompletableFuture.completedFuture(null);
+            }
+            return sendTopicPolicyEvent(topicName, ActionType.DELETE, null);
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java:141]: please add a brief code comment about the reason for this logic.

**Review by lhotari — CHANGES_REQUESTED** [review]
Please check the review comments.

File: bin/ledgers.md
Comment (poorbarcode) [reply bin/ledgers.md:1]: Removed, thanks for reminding this

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java (hunk 1 above)
Comment (poorbarcode) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java]: Added brief code comment. And I found a bug `if (topicExistsInfo.isExists()) -> if (!topicExistsInfo.isExists())` ❤️

**Review by lhotari — APPROVED** [review]
LGTM, good work @poorbarcode

**Comment by lhotari** [comment]
Triggering CI with fix #23431

**Review by Technoboy- — APPROVED** [review]

**Comment by lhotari** [comment]
I wonder if this is causing #23474 which currently blocks Pulsar CI. There's PR #23478 to disable OneWayReplicatorTestBase tests the problem is addressed. @poorbarcode do you have a chance to fix the problem?

**Comment by poorbarcode** [comment]
> I wonder if this is causing https://github.com/apache/pulsar/issues/23474 which currently blocks Pulsar CI. There's PR https://github.com/apache/pulsar/pull/23478 to disable OneWayReplicatorTestBase tests the problem is addressed. @poorbarcode do you have a chance to fix the problem?

Self-assigned https://github.com/apache/pulsar/issues/23474, I will fix it next week

**Comment by lhotari** [comment]
@poorbarcode Please also handle cherry-picking to branch-3.0 and branch-3.3

**Comment by heesung-sn** [comment]
It appears that this PR failed the test in branch-3.3.

https://github.com/heesung-sn/pulsar/actions/runs/11535492430/job/32111441443

**Comment by poorbarcode** [comment]
@heesung-sn 

> It appears that this PR fails the test in branch-3.3.

Sure, it will be fixed by the PR: https://github.com/apache/pulsar/pull/23522

**Comment by lhotari** [comment]
@poorbarcode Do you have a chance to take a look at #23543 ? It's a deadlock in SystemTopicBasedTopicPoliciesService which happens also after #23522 has been merged. The deadlock might be completely unrelated to the recent changes, but since you know SystemTopicBasedTopicPoliciesService well, I'd appreciate it if you can take a look at the thread dump and the deadlocks.

**Pull request merged**